### ISSUE TITLE
Style/RedundantSelf creates false positive with parallel assignment

### ISSUE THREAD

**Issue by daniel-rikowski** [bbatsov/rubocop#4197]
When using parallel assignment involving `self` Style/RedundantSelf thinks `self` is redundant.
Expanding the parallel assignment works, i.e. `self` is recognized as non-redundant:

With parallel assignment:

```ruby
old_log_level, self.log_level = log_level, :fatal  # Redundant `self` detected
```

Serial assignment:
```ruby
old_log_level = log_level
self.log_level = :fatal # No redundant `self` detected
```

--------

## Expected behavior

Rubocop should notice that `self` is not redundant.

## Actual behavior

Rubocop thinks that `self` is redundant.

## Steps to reproduce the problem

Check this code with Style/RedundantSelf enabled:

```ruby
def parallel
  old_log_level, self.log_level = log_level, :fatal
end

def serial
  old_log_level = log_level
  self.log_level = :fatal
end
```
## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.3 i386-mingw32)
```
